app/services/extension.js

```diff
@@ -135,9 +137,14 @@ export default Service.extend({
 
   /* Schema Editor */
   validateSchema() {
-    get(this, 'data._schema').forEach(schemaField => {
-      set(schemaField, 'validation', validateSchemaField(schemaField));
-    });
+    const allValid = get(this, 'data._schema').map(schemaField => {
+      const validation = validateSchemaField(schemaField);
+      set(schemaField, 'validation', validation);
+
+      return (validation.length === 0)
+    }).every(Boolean);
```

Comment:
So the argument for  `.every()` should actually be a function that takes the array element as an argument and returns a boolean.  
e.g.
```
var arr = [0,1,2,3];
arr.every(item => typeof item == 'number'); // => true
arr.every(item => item < 2); // => false
```

Reply:
So, since you're doing two things here
**Option A**  (with .every())
```
get(this, 'data._schema').forEach(schemaField => {
    set(schemaField, 'validation', validateSchemaField(schemaField));
});
 const allValid = get(this, 'data._schema').every(schemaField => validateSchemaField(schemaField).length === 0);
```

Or to prevent looping through the array an extra time
```
let allValid = true;
get(this, 'data._schema').forEach(schemaField => {
    const validation = validateSchemaField(schemaField);
    set(schemaField, 'validation', validation);
    if (allValid && validation.length !== 0) allValid = false;
});

```